strawberryfields/api/job.py

```diff
@@ -119,14 +121,26 @@ def result(self) -> Result:
             )
         return self._result
 
+    @property
+    def meta(self) -> dict:
+        """Returns a dictionary of metadata on job execution, such as error
+        details.
+
+        Returns:
+            dict[str, str]
+        """
+        return self._meta
+
     def refresh(self):
-        """Refreshes the status of an open or queued job,
+        """Refreshes the status and metadata of an open or queued job,
         along with the job result if the job is newly completed.
         """
         if self._status.is_final:
             self.log.warning("A %s job cannot be refreshed", self._status.value)
             return
-        self._status = JobStatus(self._connection.get_job_status(self.id))
+        job_info = self._connection.get_job(self.id)
+        self._status = JobStatus(job_info.status)
+        self._meta = job_info.meta
```

Comment:
Would it be a good idea to log the retrieved metadata at the DEBUG level here? Combined with #360, this will allow retrieved job metadata to appear within the users logs (if they decide to configure it).

Reply:
Good idea, this has been added! Let me know if you think the message formatting needs to be changed.